### PR TITLE
List choices for choice fields as enums in Swagger openapi spec

### LINKED ISSUE #4062: List ChoiceField choices in Swagger as enum
### Environment
* Python version:  3.7.3
* NetBox version:  2.7.2

### Proposed Functionality

Based on this specification https://swagger.io/docs/specification/data-models/enums/ the ChoiceField choices for label and value can (and should) be listed as enums.

### Use Case

Proper generation of API model from openapi specification for 3rd party projects without requirement to query OPTIONS for each endpoint to receive extra data from django. For example we are generating Java API for Netbox based on openapi spec from Netbox. Also having the choices listed on api/docs Swagger UI is very helpful when browsing through the API docs.

### Database Changes

None

### External Dependencies

None

### PR DESCRIPTION
**Pull request by deathbeam**
### Fixes #4062

Signed-off-by: Tomas Slusny <slusnucky@gmail.com>

### CODE REVIEW

**Review by kobayashi — CHANGES_REQUESTED** [review]
@deathbeam Could you rebase this with the latest develop branch?
The following place also needs change for your fix.
https://github.com/netbox-community/netbox/blob/f7c6df6e6add402b5265a7252ec5eaef9bcb0b9f/netbox/utilities/custom_inspectors.py#L93-L95

**Comment by deathbeam** [comment]
> @deathbeam Could you rebase this with the latest develop branch?
> The following place also needs change for your fix.
> https://github.com/netbox-community/netbox/blob/f7c6df6e6add402b5265a7252ec5eaef9bcb0b9f/netbox/utilities/custom_inspectors.py#L93-L95

Thanks, updated.

File: netbox/utilities/custom_inspectors.py [hunk 1]
```diff
@@ -76,26 +76,28 @@ def field_to_swagger_object(self, field, swagger_object_type, use_references, **
...
                 # - face is an integer set {0, 1} which is easily confused with {False, True}
                 schema_type = openapi.TYPE_STRING
-                if all(type(x) == bool for x in [c for c in choices if c is not None]):
+                if all(type(x) == bool for x in [c for c in choice_value if c is not None]):
                     schema_type = openapi.TYPE_BOOLEAN
-                value_schema = openapi.Schema(type=schema_type)
+                value_schema = openapi.Schema(type=schema_type, enum=choice_value)
                 value_schema['x-nullable'] = True
 
             if isinstance(choices[0], int):
```
Comment (kobayashi) [inline netbox/utilities/custom_inspectors.py]: `choice_value` should be `choice_value` .

**Review by kobayashi — CHANGES_REQUESTED** [review]

File: netbox/utilities/custom_inspectors.py (hunk 1 above)
Comment (deathbeam) [reply netbox/utilities/custom_inspectors.py]: Updated

**Review by kobayashi — APPROVED** [review]
LGTM 👍

**Comment by kobayashi** [comment]
Thanks @deathbeam for this change!

**Pull request merged**